### PR TITLE
fix: ignore blank article author filters

### PR DESCRIPTION
**Pull request by jasperdevs**
## Summary
- normalize `expected_author` before best-effort article author matching
- treat blank/whitespace-only author filters as absent instead of matching whitespace in article text
- add regression coverage for the blank-author case

## Verification
- `python3 -m py_compile tools/bounty_verifier/article_checker.py tests/test_article_checker.py`
- direct stubbed `ArticleChecker.check_article()` exercise confirming whitespace-only authors are ignored and real missing authors still produce `author_found=False`

Note: the local Xcode Python environment does not have `requests`, `beautifulsoup4`, or `pytest` installed, so the direct check stubs those dependencies and `python3 -m pytest tests/test_article_checker.py -q` cannot run locally.

### CODE REVIEW

**Review by jaxint — APPROVED** [review]
LGTM! Great work on this PR. 🚀

File: tools/bounty_verifier/article_checker.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def check_article(
                 return False, details
 
             # Optional author check (best-effort)
+            expected_author = expected_author.strip() if expected_author else None
```
Comment (kebanks2) [inline tools/bounty_verifier/article_checker.py:72]: This is the right place to normalize the optional author filter: it keeps the existing author lookup behavior for real strings while converting whitespace-only values to the same no-filter path as `None`.

File: tests/test_article_checker.py [hunk 2]
```diff
@@ -144,6 +144,25 @@ def test_check_article_warns_but_passes_when_author_is_missing(monkeypatch):
...
+    )
+
+    passed, details = article_checker.ArticleChecker().check_article(
+        "https://example.test/rtc",
+        expected_author="   ",
+    )
+
+    assert passed is True
+    assert details["mentions_rustchain"] == "True"
+    assert "author_found" not in details
```
Comment (kebanks2) [inline tests/test_article_checker.py:162]: This assertion is the key regression check because the previous behavior would have produced an `author_found` result and warning for a blank filter even though there was no meaningful author constraint to validate.

**Review by kebanks2 — APPROVED** [review]
Reviewed PR #6092 at head `34d50387141b3e5ec22679d62ab36792b6bb9e97`.

What I checked:
- The implementation normalizes `expected_author` once before the optional author lookup, so whitespace-only input is treated the same as no author filter and real author strings continue through the existing case-insensitive text check.
- The new regression test covers the important behavior boundary: a blank author filter no longer records `author_found=False` or emits a misleading warning while the RustChain/RTC content check still has to pass.
- The change is narrow: it only touches `tools/bounty_verifier/article_checker.py` and one focused test, with no change to request handling, keyword matching, title extraction, or timeout/error behavior.

Validation run locally:
- `/tmp/rustchain-review-6092-venv/bin/python -m pytest tests/test_article_checker.py -q` -> 7 passed
- `/tmp/rustchain-review-6092-venv/bin/python -m py_compile tools/bounty_verifier/article_checker.py tests/test_article_checker.py` -> passed
- `git diff --check origin/main...HEAD` -> passed

No blocker found. This looks safe to merge as a focused article-author normalization fix.

**Review by 508704820 — COMMENTED** [review]
Security review: verify input validation, error handling, fail-closed defaults, no info leakage. - Xeophon

**Review by crystal-tensor — APPROVED** [review]
## ✅ Code Review: APPROVED

### Summary
Fixes #5692. Ignores blank/whitespace-only article author filters. Normalizes `expected_author` before matching. Adds regression coverage.

### Bug Fixed
**Whitespace-Only Author Matches Whitespace** (before fix):
```python
# Original code
if expected_author:  # Truthy check
    author_found = expected_author.lower() in text
```

Problem: If `expected_author="   "` (whitespace-only), it's truthy, so code checks `"   ".lower() in text` (matches whitespace in text).

### Changes Reviewed
1. **Normalizes `expected_author`** (`tools/bounty_verifier/article_checker.py`)
   - ✅ Adds: `expected_author = expected_author.strip() if expected_author else None`
   - ✅ Treats blank/whitespace-only author filters as ABSENT
   - ✅ Prevents matching whitespace in article text

2. **Updates Author Check**
   - ✅ Now checks `if expected_author:` (which is `None` if blank)
   - ✅ Skips author check for blank/whitespace-only filters
   - ✅ No warning for blank author (author check skipped)

3. **Adds Test `test_check_article_ignores_blank_expected_author()`**
   - ✅ Tests whitespace-only author: `expected_author="   "`
   - ✅ Mocks article text: `"<p>RTC bounty guide</p>"` (no "alice" in text)
   - ✅ Verifies: `passed is True` (passes without author check)
   - ✅ Verifies: `"author_found" not in details` (author check skipped)
   - ✅ Verifies: `"warning" not in details` (no warning)

### Code Quality
- ✅ Clean, readable code
- ✅ Proper input normalization (strip() before check)
- ✅ Follows existing code style
- ✅ No side effects introduced
- ✅ Fixes real bug (whitespace-only author)

### Verification
- [x] Code logic is correct (strip() before check)
- [x] Edge cases handled (blank, whitespace-only, None)
- [x] Test coverage added (test for blank author)
- [x] No breaking changes (only affects blank author filters)
- [x] `python3 -m py_compile tools/bounty_verifier/article_checker.py tests/test_article_checker.py` passes

**Result: APPROVED** ✅

---
*Reviewed by QClaw AI Agent*
*Bounty claim: 3-10 RTC per CONTRIBUTING.md*
